taichi/backends/opengl/opengl_api.cpp

```diff
@@ -490,23 +485,33 @@ struct GLSLRuntime {
 struct GLSLLauncherImpl {
   std::unique_ptr<GLSSBO> root_ssbo;
   std::unique_ptr<GLSSBO> runtime_ssbo;
+  std::unique_ptr<GLSSBO> gtmp_ssbo;
   std::vector<GLSSBO> ssbo;
   std::vector<char> root_buffer;
+  std::vector<char> gtmp_buffer;
   std::unique_ptr<GLSLRuntime> runtime;
   std::vector<std::unique_ptr<CompiledProgram>> programs;
 };
 
-GLSLLauncher::GLSLLauncher(size_t size) {
+GLSLLauncher::GLSLLauncher(size_t root_size) {
   initialize_opengl();
   impl = std::make_unique<GLSLLauncherImpl>();
-  impl->root_ssbo = std::make_unique<GLSSBO>();
+
   impl->runtime_ssbo = std::make_unique<GLSSBO>();
   impl->runtime = std::make_unique<GLSLRuntime>();
-  impl->root_buffer.resize(size, 0);
-  impl->root_ssbo->bind_data(impl->root_buffer.data(), size);
-  impl->root_ssbo->bind_index(0);
   impl->runtime_ssbo->bind_data(impl->runtime.get(), sizeof(GLSLRuntime));
   impl->runtime_ssbo->bind_index(6);
+
+  impl->root_ssbo = std::make_unique<GLSSBO>();
+  impl->root_buffer.resize(root_size, 0);
+  impl->root_ssbo->bind_data(impl->root_buffer.data(), root_size);
+  impl->root_ssbo->bind_index(0);
+
+  size_t gtmp_size = 32 * 1024;  // 32 KiB, big enough
```

Comment:
Please use this one: https://github.com/taichi-dev/taichi/blob/97e1b503e55eb465a909a123014d6d935c3d9758/taichi/inc/constants.h#L9

Reply:
t4yi, I'll do a TPRT on this to decide.

Reply:
SGTM, but since it's bound once at compile time, i don't know why the size could matter here.. Anyway, can you then make `gtmp_size` a `constexpr`?